File: demo-simple-java/src/main/java/nl/dionsegijn/simple_demo/MainActivity.java
package nl.dionsegijn.simple_demo;

import android.graphics.drawable.Drawable;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import java.util.concurrent.TimeUnit;

import nl.dionsegijn.konfetti.KonfettiView;
import nl.dionsegijn.konfetti_core.NewEmitter.Emitter;
import nl.dionsegijn.konfetti_core.NewEmitter.EmitterConfig;
import nl.dionsegijn.konfetti_core.Party;
import nl.dionsegijn.konfetti_core.PartyFactory;
import nl.dionsegijn.konfetti_core.Position;
import nl.dionsegijn.konfetti_core.models.Shape;
import nl.dionsegijn.konfetti_core.models.Size;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final Drawable drawable = ContextCompat.getDrawable(getApplicationContext(), R.drawable.ic_heart);
        final Shape.DrawableShape drawableShape = new Shape.DrawableShape(drawable, true);

        final KonfettiView konfettiView = findViewById(R.id.konfettiView);
        EmitterConfig emitterConfig = new Emitter(5L, TimeUnit.SECONDS).perSecond(50);
        Party party = new PartyFactory(emitterConfig)
                .angle(359)
                .spread(90)
                .setStartVelocity(1f, 5f)
                .timeToLive(2000L)
                .shapes(new Shape.Rectangle(0.2f), drawableShape)
                .sizes(new Size(12, 5f))
                .position(new Position.relative(0.0, 0.0).between(new Position.relative(1.0, 0.0)))
                .build();
        konfettiView.setOnClickListener(view ->
                konfettiView.start(party)
        );
    }
}

File: demo-simple-java/src/main/java/nl/dionsegijn/simple_demo/MainActivity.kt
package nl.dionsegijn.simple_demo

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View

import nl.dionsegijn.konfetti.KonfettiView
import nl.dionsegijn.konfetti.models.Shape
import nl.dionsegijn.konfetti.models.Size

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val konfettiView = findViewById<KonfettiView>(R.id.konfettiView)
        konfettiView.setOnClickListener {
            konfettiView.build()
                .addColors(Color.YELLOW, Color.GREEN, Color.MAGENTA)
                .setDirection(0.0, 359.0)
                .setSpeed(1f, 5f)
                .setFadeOutEnabled(true)
                .setTimeToLive(2000L)
                .addShapes(Shape.RECT, Shape.CIRCLE)
                .addSizes(Size(12, 5f))
                .setPosition(-50f, konfettiView.width + 50f, -50f, -50f)
                .streamFor(300, 5000L)
        }

        findViewById<View>(R.id.demo_bitmap).setOnClickListener {
            startActivity(Intent(this@MainActivity, BitmapDemoActivity::class.java))
        }
    }
}
